### DISCUSSION TITLE
Usage with MSW Data

### DISCUSSION THREAD

**Discussion by ericchernuka** [valendres/playwright-msw#23] [Q&A]
~We're currently using `@msw/data` for our data modelling of the endpoints, and I'm wondering if anyone has tried using it in conjunction with `playwright-msw`? Whenever the route is initialized, it seems empty even if I attempt to seed it. I'm assuming this is due to the contexts in which its being run that's causing the issue. Any thoughts would be great.~

Update: Turned out to be entirely a bug in our code.

**Comment by valendres** [comment]
Hey @ericchernuka, sorry for the delay. Glad to hear that you've got this sorted. I personally haven't used @msw/data before, but I'd be interested to know how you're using it with playwright-msw, and what issue/bug you ran into that caused the integration to not work so well. I'm thinking that it may be helpful for others in the future 🙂

**Reply by ericchernuka** [reply]
You bet. We have an option built into our app using environment variables to dictate if we should use the MSW mock endpoint or skip it. This is useful when we intentionally limit the mocking when developing. Sadly we didn't have an option that handled a particular use case we had, resulting in no endpoints being served.  In addition, we had a `seed()` function embedded in our browser worker file, which was confusing as to why we couldn't override the endpoint with specific data.

Sadly it all came down to many assumptions about how we'd use MSW and MSW/data before doing playwright, which resulted in weirdness. Once we resolved all those, everything worked great.
